plugins/modules/proxmox_storage_list.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright Julian Vanden Broeck (@l00ptr) <julian.vandenbroeck at dalibo.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+DOCUMENTATION = """
+---
+module: proxmox_storage_list
```

Comment:
The module should be called `xxx_info`, not `xxx_list`.
```suggestion
module: proxmox_storage_info
```

Reply:
proxmox_storage_info already exists, in our case we want to list the content of a storage. So why do you think we should rename it ?

Reply:
Hmm, I just looked at that module, and I'm not sure I understand the differences between your module and the other module. Both modules allow to list storage. Why not extend the other module to allow to return the information you want this new module to return?

Reply:
"my" module allows to list the content within a specific storage. So imagine we want to list all VMs, backups, isos,... we can use `proxmox_storage_list`.

With this new module you can get that kind of listing:
```
        "proxmox_storages_content": [
            {
                "content": "backup",
                "ctime": 1648406794,
                "format": "pbs-ct",
                "size": 7039562175,
                "subtype": "lxc",
                "verification": {
                    "state": "ok",
                    "upid": "UPID:datastore1:00130F49:1A12D8375:00001A65:65738AD8:verificationjob:daily_pra\\x3av\\x2dd0cc18c5\\x2d8707:root@pam:"
                },
                "vmid": 1018,
                "volid": "datastore:backup/ct/1018/2022-03-27T18:46:34Z"
            },
            {
                "content": "backup",
                "ctime": 1650825932,
                "format": "pbs-ct",
                "size": 7109029332,
                "subtype": "lxc",
                "verification": {
                    "state": "ok",
                    "upid": "UPID:datastore1:00130F49:1A12D8375:00001900:656F9658:verificationjob:daily_pra\\x3av\\x2dd0cc18c5\\x2d8707:root@pam:"
                },
                "vmid": 1018,
                "volid": "datastore:backup/ct/1018/2022-04-24T18:45:32Z"
            },
            {
                "content": "backup",
                "ctime": 1653850280,
                "format": "pbs-ct",
                "size": 7216809280,
                "subtype": "lxc",
                "verification": {
                    "state": "ok",
                    "upid": "UPID:datastore1:00130F49:1A12D8375:00001AFB:6574DC58:verificationjob:daily_pra\\x3av\\x2dd0cc18c5\\x2d8707:root@pam:"
                },
                "vmid": 1018,
                "volid": "datastore:backup/ct/1018/2022-05-29T18:51:20Z"
            }]
```
I would like to maintain and use that new module to list content of some backup storage attached to my clusters and restore automatically them. So my use case is that one:
- Get `backup` on a specific `storage` for a `vmid`
- Parse the result (or improve that module to caculate) to get the newest backup
- then use another task / module to create a VM / LXC based on that backup (using `volid`)


With `proxmox_storage_info` you get the "only" the metadata, basically the kind of object / `content` you can store, nodes attached to that storage, if it's `shared` or not, the `type` (nfs, lvm,...),..

For your information proxmox_storage_info returns those data:
```
        "proxmox_storages": [
            {
                "content": [
                    "backup"
                ],
                "datastore": "daily_pra",
                "digest": "8b1f177e24b4bda92e4c113003180bdae3716e61",
                "fingerprint": "xx:xx:xx:xx",
                "prune-backups": {
                    "keep-all": "1"
                },
                "server": "filer.example.net",
                "shared": true,
                "storage": "datastore",
                "type": "pbs",
                "username": "user@pbs"
            }
        ]
```

Implementing `proxmox_storage_list` on `proxmox_storage_info` will increase the complexity of the functions / methods used to get those metadata. IMHO If we move in that direction, we will have to handle a new flag to decide which API call we should do (we don't use the same endpoint for those two modules)  and probably add some more complex test cases.

Reply:
@felixfontein do you think i should implement that on proxmox_storage_info ?

Reply:
Thanks for the clarifications. I agree that having two separate modules is better.

How about naming this module `proxmox_storage_contents_info`?

()

Reply:
your proposition seems better than my initial proposal :) let's rename to `proxmox_storage_contents_info`.

Reply:
History is becoming messy, if you agree @felixfontein  i will rebase / squash all those fixup commit.

Reply:
Sure, feel free. We merbe by squashing anyway, but reducing the number of commits also improves the resulting commit message.

Reply:
squashed all that fixup commits and also the 2 first commits. I also fixed the few typo (adding missing dot and replacing `LXC` by `lxc`)